### ISSUE TITLE
Save states fail to load

### ISSUE THREAD

**Issue by zcal** [libretro/RetroArch#12374]
## Description

Save states fail to load.

### Expected behavior

The running content should be re-loaded in the saved state.

### Actual behavior

No error is thrown in the console.  Neither the core nor RetroArch crash.  RetroArch falsely reports via notification that the save state was loaded.  The content continues to be run in the state it was in prior to loading the save state.

I've tested this on two different cores with the same result.

### Steps to reproduce the bug

1. Load content for which you have an existing save state.
2. Once the content is running, open the RetroArch menu and attempt to load your save state.
3. Notice that RetroArch fails to load the save state, but reports via notification that it was loaded.

### Bisect Results

This behavior began on upgrade to RetroArch 1.9.2

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.9.2
- Git Version: bc23d070de

### Environment information

- OS: Arch Linux
- Compiler: installed from Community package repository

**Comment by zcal** [comment]
Just downgraded to RetroArch 1.9.0 and I was surprised to find that I still experience this issue.  Installed 1.9.1 and it doesn't work there, either.

I checked my directories configuration to be sure that states were being loaded from the correct place.  They are.

I know save states were working for me the day prior to updating to 1.9.2 (was using 1.9.1).  I've changed nothing about my RetroArch configuration, so I'm not convinced this isn't a bug.  Perhaps a bug with a dependency?

**Comment by bslenul** [comment]
Happening with every core and every game? Sure you got the correct slot number selected in Quick Menu? If the extension of your state file is .state then it should be slot 0, if it's .state1 then slot 1, etc.

**Comment by zcal** [comment]
> Happening with every core and every game? Sure you got the correct slot number selected in Quick Menu? If the extension of your state file is .state then it should be slot 0, if it's .state1 then slot 1, etc.

Yes and yes.  Although I have confirmed that newly saved states will load properly.  The problem affects all of my save states created prior to upgrading to RetroArch 1.9.2.

For good measure, I did check my system backups in case there was some data corruption, but my backed up states get the same checksums, so nothing's changed about the files themselves.
